integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/JdbcHandler.java

```diff
@@ -39,23 +44,57 @@ public boolean isClass(TableCatalog tableCatalog) {
     return tableCatalog instanceof JDBCTableCatalog;
   }
 
-  @SneakyThrows
   @Override
   public DatasetIdentifier getDatasetIdentifier(
       SparkSession session,
       TableCatalog tableCatalog,
       Identifier identifier,
       Map<String, String> properties) {
-    JDBCTableCatalog catalog = (JDBCTableCatalog) tableCatalog;
-    JDBCOptions jdbcOptions = (JDBCOptions) FieldUtils.readField(catalog, "options", true);
+    Optional<JDBCOptions> jdbcOptions = getJdbcOptions(tableCatalog);
 
+    if (!jdbcOptions.isPresent()) {
+      throw new MissingDatasetIdentifierCatalogException(
+          "No jdbc options found for catalog " + tableCatalog.name());
+    }
+
+    JDBCOptions options = jdbcOptions.get();
     List<String> parts =
         Stream.concat(Arrays.stream(identifier.namespace()), Stream.of(identifier.name()))
             .collect(Collectors.toList());
 
     return namespaceResolver.resolve(
         JdbcDatasetUtils.getDatasetIdentifier(
-            jdbcOptions.url(), parts, jdbcOptions.asConnectionProperties()));
+            options.url(), parts, options.asConnectionProperties()));
+  }
+
+  @Override
+  public Optional<OpenLineage.CatalogDatasetFacet> getCatalogDatasetFacet(
```

Comment:
Does it make sense to add this facet for JDBCRelation? There is no catalog registered in Spark session